test/CMakeLists.txt

```diff
@@ -18,7 +18,42 @@ set(TOPICS
   Array2m
 )
 
-function(add_performance_test TEST_NAME COMM RMW_IMPLEMENTATION)
+function(add_performance_test TEST_NAME COMM RMW_IMPLEMENTATION SYNC_MODE)
+
+  if(${SYNC_MODE} STREQUAL "async" AND ${RMW_IMPLEMENTATION} STREQUAL "rmw_cyclonedds_cpp")
+    message(STATUS "No async mode in RMW_IMPLEMENTATION: rmw_cyclonedds_cpp")
+    return()
+  endif()
+
+  if(${SYNC_MODE} STREQUAL "sync" AND ${RMW_IMPLEMENTATION} STREQUAL "rmw_opensplice_cpp")
+    message(STATUS "No sync mode in RMW_IMPLEMENTATION: rmw_opensplice_cpp")
+    return()
+  endif()
+
+  if(${SYNC_MODE} STREQUAL "sync" AND ${RMW_IMPLEMENTATION} STREQUAL "rmw_connext_cpp")
+    message(STATUS "No sync mode in RMW_IMPLEMENTATION: rmw_connext_cpp")
+    return()
+  endif()
+
+  # Enviroment variables to configure rmw_fastrtps_cpp in sync or async
+  if(${SYNC_MODE} STREQUAL "sync" AND
+    (${RMW_IMPLEMENTATION} STREQUAL "rmw_fastrtps_cpp" OR ${RMW_IMPLEMENTATION} STREQUAL "rmw_fastrtps_dynamic_cpp"))
+      set(RMW_FASTRTPS_USE_QOS_FROM_XML 1)
+      get_filename_component(
+          FASTRTPS_DEFAULT_PROFILES_FILE
+          "${CMAKE_CURRENT_SOURCE_DIR}/DEFAULT_FASTRTPS_PROFILES.xml"
+          ABSOLUTE
+      )
+      get_filename_component(
+          FASTRTPS_DEFAULT_PROFILES_TWO_PROCESSES_FILE
+          "${CMAKE_CURRENT_SOURCE_DIR}/DEFAULT_FASTRTPS_PROFILES_TWO_PROCESSES.xml"
+          ABSOLUTE
+      )
+  else()
+    set(RMW_FASTRTPS_USE_QOS_FROM_XML 0)
+    set(FASTRTPS_DEFAULT_PROFILES_FILE "")
+    set(FASTRTPS_DEFAULT_PROFILES_TWO_PROCESSES_FILE "")
```

Comment:
I was testing this and I got this log from Fast-RTPS
```log
2020-03-17 08:09:58.569 [XMLPARSER Error] Error loading XML file, filename empty -> Function loadXMLFile
```

The error does no affect the middleware behavior, but I will pollute your output twice (one for pub and 1 for sub).  I was thinking that the parser could drop all the lines before `---EXPERIMENT-START---` (you are skipping them anyway). Since the logs are generated when creating the participants, they should always be above that.